code/game/gamemodes/cult/blood_magic.dm

```diff
@@ -529,10 +529,10 @@
 	if(QDELETED(src) || !user || !user.is_holding(src) || user.incapacitated() || !actual_selected_rune)
 		return
 
-	if(HAS_TRAIT(user, TRAIT_FLOORED))
-		to_chat(user, "<span class='cultitalic'>You cannot cast this spell while knocked down!</span>")
-		return
-
+	if(user == target)
+		if(!do_after(user, 3 SECONDS, allow_moving = TRUE, allow_moving_target = TRUE, target = teleportee))
```

Comment:
I still suggest you do this, but you'll have to go to `mob_helpers.dm:470` and introduce new arguments and pass them into the original function
```suggestion
		if(!do_after_once(user, 3 SECONDS, allow_moving = TRUE, allow_moving_target = TRUE, target = teleportee))
```
```
/proc/do_after_once(mob/user, delay, needhand = 1, atom/target = null, progress = 1, allow_moving, must_be_held, use_default_checks, allow_moving_target, attempt_cancel_message = "Attempt cancelled.", special_identifier)
	if(!user || !target)
		return

	var/cache_key = "[user.UID()][target.UID()][special_identifier]"
	if(GLOB.do_after_once_tracker[cache_key])
		GLOB.do_after_once_tracker[cache_key] = DOAFTERONCE_MAGIC
		to_chat(user, "<span class='warning'>[attempt_cancel_message]</span>")
		return FALSE
	GLOB.do_after_once_tracker[cache_key] = TRUE
	. = do_after(user, delay, needhand, target, progress, allow_moving, must_be_held, list(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(do_after_once_checks),  cache_key)), use_default_checks, allow_moving_target)
	GLOB.do_after_once_tracker[cache_key] = FALSE
```